src/rgw/rgw_rest_conn.cc

```diff
@@ -358,10 +424,19 @@ int RGWRESTConn::get_resource(const DoutPrefixProvider *dpp,
   ret = req.send_request(dpp, &key, headers, resource, mgr, send_data);
   if (ret < 0) {
     ldpp_dout(dpp, 5) << __func__ << ": send_request() resource=" << resource << " returned ret=" << ret << dendl;
+    if (ret == -EIO)
+      set_url_unconnectable(url);
```

Comment:
style nit: please use curly-braces even for single statements, see https://github.com/ceph/ceph/blob/main/CodingStyle#L84-L103